### PR TITLE
Add authorizations-related endpoints

### LINKED ISSUE #4766: [Authorizations] Fetch single authorization and summary data
### Description

Fetch single authorization data from WCPay server's endpoint. 

### Acceptance criteria

* Create client side data endpoint to fetch authorization data from WCPay server

### Dev notes

You can check WCPay server endpoint implementation in this [PR](2465-gh-Automattic/woocommerce-payments-server)

### PR DESCRIPTION
**Pull request by elazzabi**
Fixes #4766

#### Changes proposed in this Pull Request

Title: Add authorizations-related endpoint (summary, a list of authorizations, and one single authorization)

Description: This change will make it easier to query authorizations from the server and display it on the front end. These routes aren't used yet, so nothing should break :)

#### Testing instructions

These endpoints are only wrappers for server-side endpoints. No testing is necessary here.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: includes/wc-payment-api/class-wc-payments-api-client.php [hunk 1]
```diff
@@ -2562,4 +2563,48 @@ private function get_fingerprint_metadata(): array {
...
+	 *
+	 * @param int    $page       The requested page.
+	 * @param int    $page_size  The size of the requested page.
+	 * @param string $sort       The column to be used for sorting.
+	 * @param string $direction  The sorting direction.
+	 *
+	 * @return array
+	 * @throws API_Exception - Exception thrown on request failure.
+	 */
+	public function list_authorizations( $page = 0, $page_size = 25, $sort = 'created', $direction = 'desc' ) {
```
Comment (mgascam) [inline includes/wc-payment-api/class-wc-payments-api-client.php]: The methods of this class are tested in `tests/unit/wc-payment-api/test-class-wc-payments-api-client.php`. I'd suggest to add some test cases there for the new methods we are adding. What do you think?

Comment (elazzabi) [reply includes/wc-payment-api/class-wc-payments-api-client.php]: I was trying to find where tests are written but couldn't. Thank you for the help, I'll work on tests now!

File: includes/admin/class-wc-rest-payments-authorizations-controller.php [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+			]
+		);
+	}
+
+	/**
+	 * Retrieve authorizations to respond with via API.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 */
+	public function get_authorizations( $request ) {
```
Comment (mgascam) [inline includes/admin/class-wc-rest-payments-authorizations-controller.php]: Can you add type hints here and in other function definitions in the PR please? i.e. 
```suggestion
	public function get_authorizations( WP_REST_Request $request ) {
```

Comment (elazzabi) [reply includes/admin/class-wc-rest-payments-authorizations-controller.php]: Add to two functions and removed from one given it's not used 4d98d098

**Review by mgascam — APPROVED** [review]
LGTM!

File: includes/wc-payment-api/class-wc-payments-api-client.php [hunk 3]
```diff
@@ -2560,4 +2561,48 @@ private function get_fingerprint_metadata(): array {
...
```
Comment (kalessil) [inline includes/wc-payment-api/class-wc-payments-api-client.php]: Can we add types declarations here, please?

File: includes/wc-payment-api/class-wc-payments-api-client.php [hunk 4]
```diff
@@ -2560,4 +2561,48 @@ private function get_fingerprint_metadata(): array {
...
+		return $this->request( [], self::AUTHORIZATIONS_API . '/summary', self::GET );
+	}
+
+	/**
+	 * Fetch a single authorizations with provided payment intent id.
+	 *
+	 * @param string $payment_intent_id id of requested transaction.
+	 * @return array authorization object.
+	 */
+	public function get_authorization( $payment_intent_id ) {
```
Comment (kalessil) [inline includes/wc-payment-api/class-wc-payments-api-client.php]: Can we add types declarations here, please?

File: tests/unit/wc-payment-api/test-class-wc-payments-api-client.php [hunk 5]
```diff
@@ -1987,6 +1987,99 @@ public function test_get_document_error() {
...
+
+		$this->set_http_mock_response(
+			200,
+			[
+				'payment_intent_id' => $payment_intent_id,
+			]
+		);
+
+		$authorization = $this->payments_api_client->get_authorization( $payment_intent_id );
+		$this->assertEquals( $payment_intent_id, $authorization['payment_intent_id'] );
```
Comment (kalessil) [inline tests/unit/wc-payment-api/test-class-wc-payments-api-client.php]: Strict asserts, please.

File: tests/unit/wc-payment-api/test-class-wc-payments-api-client.php [hunk 6]
```diff
@@ -1987,6 +1987,99 @@ public function test_get_document_error() {
...
+	public function test_get_authorization_not_found() {
+		$payment_intent_id = 'pi_123smtm';
+		$error_code        = 'authorization_missing';
+		$error_message     = 'The authorization you asked for does not exist';
+
+		$this->set_http_mock_response(
+			404,
+			[
+				'error' => [
+					'code'    => $error_code,
```
Comment (kalessil) [inline tests/unit/wc-payment-api/test-class-wc-payments-api-client.php]: I think we can inline the value here.

**Review by kalessil — CHANGES_REQUESTED** [review]
Look good overall, leaving some nitpicks.

File: tests/unit/wc-payment-api/test-class-wc-payments-api-client.php (hunk 6 above)
Comment (elazzabi) [reply tests/unit/wc-payment-api/test-class-wc-payments-api-client.php]: Indeed, changed here f88b5cfd I left the other ones as variables given they are used below in the test

File: tests/unit/wc-payment-api/test-class-wc-payments-api-client.php (hunk 5 above)
Comment (elazzabi) [reply tests/unit/wc-payment-api/test-class-wc-payments-api-client.php]: Fixed b0a6b70c

File: includes/wc-payment-api/class-wc-payments-api-client.php (hunk 3 above)
Comment (elazzabi) [reply includes/wc-payment-api/class-wc-payments-api-client.php]: Thank you for reminding us of this. Fixed in f452270d

**Review by kalessil — APPROVED** [review]
LGTM!

**Pull request merged**